### PR TITLE
Fix bugs in contrast guideline and improve heuristic

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Instead of using the pixels under the entire semantics node, use the renderbox of the text widget. This shrinks the number of pixels considered and makes it less likely that background colors are assumed to be part of the text contrast.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from our [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

## Tests

I added the following tests:

- regression test for material widget

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by jonahwilliams** [comment]
cc @johnsonmh

File: packages/flutter_test/lib/src/accessibility.dart [hunk 1]
```diff
@@ -200,7 +200,7 @@ class MinimumTextContrastGuideline extends AccessibilityGuideline {
     final ByteData byteData = await tester.binding.runAsync<ByteData>(() async {
       // Needs to be the same pixel ratio otherwise our dimensions won't match the
       // last transform layer.
-      image = await layer.toImage(renderView.paintBounds, pixelRatio: 1.0);
+      image = await layer.toImage(renderView.paintBounds, pixelRatio: 1 / 3);
```
Comment (johnsonmh) [inline packages/flutter_test/lib/src/accessibility.dart]: Just curious, why the `1/3` change?  It seems like the comment above this line no longer fully holds.

**Review by johnsonmh — APPROVED** [review]
LGTM, thanks Jonah!

File: packages/flutter_test/lib/src/accessibility.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_test/lib/src/accessibility.dart]: When I used the semantics node, we ended up including the dpr transform. Now that I'm using the rect from the render box, this doesn't include the dpr transform

**Comment by jonahwilliams** [comment]
Submitting, build is currently green but dashboard may be unable to update older PRs

**Pull request merged**